disperser/batcher/batchstore/minibatch_store.go

```diff
@@ -0,0 +1,707 @@
+package batchstore
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	commondynamodb "github.com/Layr-Labs/eigenda/common/aws/dynamodb"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigenda/disperser/batcher"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	gcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/google/uuid"
+)
+
+const (
+	blobMetadataIndexName = "BlobMetadataIndex"
+	batchStatusIndexName  = "BatchStatusIndex"
+	batchSK               = "BATCH#"
+	minibatchSK           = "MINIBATCH#"
+	dispersalRequestSK    = "DISPERSAL_REQUEST#"
+	dispersalResponseSK   = "DISPERSAL_RESPONSE#"
+)
+
+type DynamoBatchRecord struct {
```

Comment:
One idea for marshaling/unmarshaling go structs.
Instead of converting back and forth and maintaining two sets of types, why don't we add ``dynamodbav:"-"`` tags on fields that don't work nicely with the marshaler (i.e. `BatchID`, `OperatorID`, etc)?

We do [this](https://github.com/Layr-Labs/eigenda/blob/master/disperser/disperser.go#L82) for blob metadata currently, and convert these types manually in the `MarshalXXXRecord` and `UnmarshalXXXRecord` methods

Reply:
Not's quite the same issue. BlobMetadata is excluded because we don't want it flattened, so it gets marshaled/unmarshaled separately. Unlike OperatorID type, The BlobMetadata struct is compatible with Marshal/UnmarshalMap.

We can use `dynamodbav:"-"` to exclude `BlobID` and `OperatorID`. Marshalling is pretty clean and straightforward, but Umarshalling requires manual iteration and is kinda ugly.

Reply:
I think cleanest solution is exclude `BlobID` and `OperatorID` via `dynamodbav:"-"` and then define custom dynamo types for them and then do multiple calls to UnmarshalMap using the base struct and then dynamo struct
```
type DynamoBatchId struct {
	BatchID string
}

type DynamoOperatorId struct {
	OperatorID string
}
```